ledger/complete/wal/checkpoint_v6_writer.go

```diff
@@ -327,37 +333,48 @@ func storeSubTrieConcurrently(
 		return nil, 0, nil, fmt.Errorf("invalid nWorker %v, the valid range is [1,%v]", nWorker, subtrieCount)
 	}
 
-	resultChs := make(chan chan *resultStoringSubTrie, nWorker-1)
-	go func() {
-		for i, subTrieRoot := range subtrieRoots {
-			resultCh := make(chan *resultStoringSubTrie)
-			resultChs <- resultCh
-			// resultsChs only has nWorker number of buffer room,
-			// if resultCh can be pushed to resultChs, then we will work on this channel
-			// otherwise, the push will be blocked until the result is finished
-			go func(i int, subTrieRoot []*node.Node) {
+	jobs := make(chan job, len(subtrieRoots))
+	resultChs := make([]<-chan *resultStoringSubTrie, len(subtrieRoots))
+
+	// push all jobs into the channel
+	for i, roots := range subtrieRoots {
+		resultCh := make(chan *resultStoringSubTrie)
+		resultChs[i] = resultCh
+		jobs <- job{
+			Index:  i,
+			Roots:  roots,
+			Result: resultCh,
+		}
```

Comment:
Need to close input channel so goroutines exit properly.
```suggestion
		}
		close(jobs)
```